explorer/websockethandlers.go

```diff
@@ -97,9 +140,54 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
 
 				case "getmempooltxs":
 					exp.MempoolData.RLock()
-					msg, err := json.Marshal(exp.MempoolData)
+					mempoolData := exp.MempoolData
 					exp.MempoolData.RUnlock()
```

Comment:
As I said before, this defeats the purpose of the lock. `exp.MempoolData` is a pointer. So when you read the data via the `mempoolData` variable, it's the same data the lock was trying to protect.  You either need a deep copy, or keep it locked until you are done with the underlying shared data.